rotkehlchen/chain/ethereum/modules/balancer/balancer.py

```diff
@@ -0,0 +1,710 @@
+from rotkehlchen.chain.ethereum.trades import AMMSwap, AMMTrade
+from rotkehlchen.constants import ZERO
+from rotkehlchen.errors import DeserializationError, RemoteError
+from rotkehlchen.fval import FVal
+from rotkehlchen.inquirer import Inquirer
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.premium.premium import Premium
+from rotkehlchen.typing import ChecksumEthAddress, Location, Price, Timestamp
+from rotkehlchen.user_messages import MessagesAggregator
+from rotkehlchen.utils.interfaces import EthereumModule
+
+from .graph import POOLSHARES_QUERY, SWAPS_QUERY, TOKENPRICES_QUERY
+from .typing import (
+    BALANCER_TRADES_PREFIX,
+    AddressBalances,
+    AddressSwaps,
+    AddressTrades,
+    DDAddressBalances,
+    DDAddressSwaps,
+    ProtocolBalance,
+    TokenPrices,
+)
+from .utils import (
+    deserialize_pool_share,
+    deserialize_swap,
+    deserialize_token_day_data,
+    deserialize_token_price,
+    get_trades_from_tx_swaps,
+)
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.manager import EthereumManager
+    from rotkehlchen.db.dbhandler import DBHandler
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+SUBGRAPH_REMOTE_ERROR_MSG = (
+    "Failed to request the Balancer subgraph due to {error_msg}. "
+    "All Balancer balances and historical queries are not functioning until this is fixed. "  # noqa: E501
+    "Probably will get fixed with time. If not report it to rotki's support channel"  # noqa: E501
+)
+
+
+class Balancer(EthereumModule):
+    """Balancer integration module
+
+    * Balancer subgraph:
+    https://github.com/balancer-labs/balancer-subgraph
+    """
+    def __init__(
+            self,
+            ethereum_manager: 'EthereumManager',
+            database: 'DBHandler',
+            premium: Optional[Premium],
+            msg_aggregator: MessagesAggregator,
+    ) -> None:
+        self.ethereum = ethereum_manager
+        self.database = database
+        self.premium = premium
+        self.msg_aggregator = msg_aggregator
+        self.trades_lock = Semaphore()
+        try:
+            self.graph: Optional[Graph] = Graph(
+                'https://api.thegraph.com/subgraphs/name/balancer-labs/balancer',
+            )
+        except RemoteError as e:
+            self.graph = None
+            self.msg_aggregator.add_error(
+                f"Could not initialize the Balancer subgraph due to {str(e)}. "
+                f"All Balancer balances and historical queries are not functioning until this is fixed. "  # noqa: E501
+                f"Probably will get fixed with time. If not report it to rotki's support channel",
+            )
+        try:
+            self.graph_uniswap: Optional[Graph] = Graph(
+                'https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v2',
+            )
+        except RemoteError as e:
+            self.graph_uniswap = None
```

Comment:
The module is usable without it, so it will remain optional. The subgraph RemoteError will be catched by `_get_unknown_token_prices_graph()` and the user will be notified with all the unknown tokens that we weren't able to request the price.